packages/dds/sequence-sequential-id/src/SharedStringWithSequentialIdFactory.ts

```diff
@@ -0,0 +1,57 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import * as MergeTree from "@fluidframework/merge-tree";
+import { IChannelAttributes, IChannelServices,
+    IFluidDataStoreRuntime, IChannelFactory } from "@fluidframework/datastore-definitions";
+import { SharedString } from "@fluidframework/sequence";
+import { pkgVersion } from "./packageVersion";
+import { createStringWithSequentialIdFactory } from "./sharedStringWithSequentialId";
+
+export class SharedStringWithSequentialIdFactory implements IChannelFactory {
+    public static Type = "https://graph.microsoft.com/types/mergeTree/sequential-id-sharedstring";
```

Comment:
i'm not sure this is what you want. back compat could be hard by creating a fully new dds

Reply:
Can you please elaborate? The way I see it is:
- clients who do not want this new behavior should not pay a cost for it (i.e. it's opt-in).
- I'm not sure we want to patch old files that do not have sequential IDs on open.

So I think we want to have a small wrapper on Bohemia side that exposes certain interface, but wraps either of the DDSs. And new files will get new DDS, while that wrapper would do something reasonable for old files with old DDS (like when asked for range of IDs, it would either give sequential IDs, or old marker IDs, and support back compat through resolving either result back to range).

Can you please describe the alternative you see being implemented?